crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Value.hs

```diff
@@ -123,6 +124,56 @@ zeroInt sym bytes k
           k (Just (blk, bv))
 zeroInt _ _ k = k @1 Nothing
 
+-- | This instance tries to make things as concrete as possible.
+instance IsExpr (SymExpr sym) => PP.Pretty (LLVMVal sym) where
+  pretty =
+    let typed doc tp = PP.text doc PP.<+> PP.text ":" PP.<+> PP.text (show tp)
+    in
+      \case
+        (LLVMValZero tp) -> PP.angles (typed "zero" tp)
+        (LLVMValUndef tp) -> PP.angles (typed "undef" tp)
+        (LLVMValInt blk w) ->
+          case asNat blk of
+            Just 0 ->
+              case (asUnsignedBV w, asSignedBV w) of
```

Comment:
Probably better to just case on `asUnsignedBV` and use `toSigned` to get the signed value instead of casing separately.